File: app/src/main/java/a84934/droidterpreter/Block.java
package a84934.droidterpreter;

import android.graphics.Rect;

public class Block {
    public BlockType type;

    public int width, height, color, deltaX, deltaY;
    public Rect r;
    public Object value;

    public enum BlockType {
        ADD,
        NUM,
        MAIN
    }

    public static BlockType fromIndex(int i){
        switch (i){
            case 0:
                return BlockType.ADD;
            case 1:
                return BlockType.NUM;
            default:
                return BlockType.NUM;
        }
    }

    public static BlockType typeFromString(String type){
        switch (type){
            case "Add":
                return BlockType.ADD;
            case "Num":
                return BlockType.NUM;
            default:
                return BlockType.NUM;
        }
    }
}

File: app/src/main/java/a84934/droidterpreter/GraphView/Block.kt
package a84934.droidterpreter.GraphView

import a84934.droidterpreter.BlockVals.BlockVals
import android.graphics.Rect

enum class BlockType {
    NUM,
    ADD,
    MULT,
    MAIN
}

fun blockTypeForIndex(i : Int) : BlockType {
    if (i < 0 || i >= BlockType.values().size){
        throw IllegalArgumentException("")
    } else {
        return BlockType.values()[i]
    }
}


class Block (val blockType : BlockType,
             val width: Int, val height: Int,
             val color: Int,
             var value: BlockVals) {

    var r: Rect = Rect()
    var deltaX: Int = 0
    var deltaY: Int = 0


}
